Open Judge System/Web/OJS.Web/Areas/Administration/Controllers/ProblemsController.cs

```diff
@@ -710,5 +735,61 @@ private void RetestSubmission(int submissionId)
             entry.Property(pr => pr.Processed).IsModified = true;
             entry.Property(pr => pr.Processing).IsModified = true;
         }
+
+        private DetailedProblemViewModel PrepareProblemViewModelForEdit(int id)
+        {
+            return this.Data.Problems.All()
+                .Where(x => x.Id == id)
+                .Select(DetailedProblemViewModel.FromProblem)
+                .FirstOrDefault();
+        }
+
+        private DetailedProblemViewModel PrepareProblemViewModelForCreate(Contest contest)
+        {
+            var lastOrderBy = -1;
+            var lastProblem = this.Data.Problems.All().Where(x => x.ContestId == contest.Id);
+
+            if (lastProblem.Any())
```

Comment:
You make 2 db calls (one for `.Any()` and one for `.Max()`) You can reduce them to only one if you get list with order by in the upper line.